collections/src/main/scala/strawman/collection/LinearSeq.scala

```diff
@@ -127,20 +127,21 @@ trait LinearSeqOps[+A, +CC[X] <: LinearSeq[X], +C <: LinearSeq[A]] extends Any w
     acc
   }
 
-  override def sameElements[B >: A](that: IterableOnce[B]): Boolean = that match {
-    case that1: LinearSeq[B] =>
-      // Probably immutable, so check reference identity first (it's quick anyway)
-      (coll eq that1) || {
-        var these: LinearSeq[A] = coll
-        var those: LinearSeq[B] = that1
-        while (!these.isEmpty && !those.isEmpty && these.head == those.head) {
-          these = these.tail
-          those = those.tail
+  override def sameElements[B >: A](that: IterableOnce[B]): Boolean = {
+    @tailrec def lazyListEq(a: LinearSeq[B], b: LinearSeq[B]): Boolean =
```

Comment:
can you rename it to `linearSeqEq` for the sake of consistency?

Reply:
Fixed.